src/Compilers/CSharp/Test/Emit2/Semantics/ParamsCollectionTests.cs

```diff
@@ -12624,6 +12600,132 @@ static void Test(params MyCollection a)
 ").VerifyDiagnostics();
         }
 
+        [Fact]
+        public void Cycle_14_ThroughCollectionBuilderAttribute()
+        {
+            var src = """
+#pragma warning disable CS0436 // The type 'CollectionBuilderAttribute' in '' conflicts with the imported type 'CollectionBuilderAttribute' in 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. Using the type defined in ''.
+
+using System;
+using System.Collections.Generic;
+
+namespace System.Runtime.CompilerServices
+{
+    [AttributeUsage(AttributeTargets.All, AllowMultiple = true)]
+    [CollectionBuilder(typeof(MyCollectionBuilder), nameof(MyCollectionBuilder.Create))]
+    public sealed class CollectionBuilderAttribute : Attribute
+    {
+        public CollectionBuilderAttribute(Type builderType, string methodName, params CollectionBuilderAttribute p) { }
+        public IEnumerator<long> GetEnumerator() => throw null;
+    }
+
+    public class MyCollectionBuilder
+    {
+        public static CollectionBuilderAttribute Create(ReadOnlySpan<long> items) => throw null;
+    }
+
+    class Program
+    {
+        static void Main()
+        {
+            Test();
+            Test(1);
+            Test(2, 3);
+        }
+
+        static void Test(params CollectionBuilderAttribute a)
```

Comment:
There's edge cases, and then there's cases that are so far off the edge that they make edge cases seem reasonable. This is the latter kind 😆.